src/getting-started.md

```diff
@@ -0,0 +1,477 @@
+
+## Cloning and Building
+
+The main repository is [`rust-lang/rust`][repo]. This contains the compiler,
+the standard library (including `core`, `alloc`, `test`, `proc_macro`, etc),
+and a bunch of tools (e.g. `rustdoc`, the bootstrapping infrastructure, etc).
+
+[repo]: https://github.com/rust-lang/rust
+
+There are also a bunch of submodules for things like LLVM, `clippy`, `miri`,
+etc. You don't need to clone these immediately, but the build tool will
+automatically clone and sync them (more later).
+
+[**Take a look at the "Suggested Workflows" chapter for some helpful
+advice.**][suggested]
+
+[suggested]: ./building/suggested.md
+
+### System Requirements
+
+[**See this chapter for detailed software requirements.**](./building/prerequisites.md)
+Most notably, you will need Python 2 or 3 to run `x.py`.
+
+There are no hard hardware requirements, but building the compiler is
+computationally expensive, so a beefier machine will help, and I wouldn't
+recommend trying to build on a Raspberry Pi :P
+
+- Recommended >=30GB of free disk space; otherwise, you will have to keep
+  clearing incremental caches. More space is better, the compiler is a bit of a
+  hog; it's a problem we are aware of.
+- Recommended >=8GB RAM.
+- Recommended >=2 cores; more cores really helps.
+- You will need an internet connection to build; the bootstrapping process
+  involves updating git submodules and downloading a beta compiler. It doesn't
+  need to be super fast, but that can help.
+
+Building the compiler takes more than half an hour on my moderately powerful
+laptop. The first time you build the compiler, LLVM will also be built unless
+you use your system's LLVM ([see below][configsec]).
+
+[configsec]: #configuring-the-compiler
+
+Like `cargo`, the build system will use as many cores as possible. Sometimes
+this can cause you to run low on memory. You can use `-j` to adjust the number
+concurrent jobs.
+
+Also, if you don't have too much free disk space, you may want to turn off
+incremental compilation ([see below][configsec]). This will make
+compilation take longer, but will save a ton of space from the incremental
+caches.
+
+### Cloning
+
+You can just do a normal git clone:
+
+```sh
+git clone https://github.com/rust-lang/rust.git
+```
+
+You don't need to clone the submodules at this time. But if you want to, you
+can do the following:
+
+```sh
+# first time
+git submodule update --init --recursive
+
+# subsequent times (to pull new commits)
+git submodule update
+```
+
+### Configuring the Compiler
+
+The compiler has a configuration file which contains a ton of settings. We will
+provide some recommendations here that should work for most, but [check out
+this chapter for more info][config].
+
+[config]: ./building/how-to-build-and-run.html#create-a-configtoml
+
+In the top level of the repo:
+
+```sh
+cp config.toml.example config.toml
+```
+
+Then, edit `config.toml`. You may want to search for, uncomment, and update
+the following settings:
```

Comment:
Without `debug = true`, you won't see any `debug!` logging. @eddyb recommended `debug = true` and `debuginfo = 1` a while back, maybe we could change debuginfo to default to 1 instead of 2 when `debug` is set?

Reply:
I've started this hackmd to collect various suggestions that have come up: https://hackmd.io/@ux-jKBcgRTSHsH042VF3BA/rkXEozKTU/edit